eduNEXT/tutor-contrib-limesurvey
# feat: enable custom configurations for LimeSurvey service

### Description
This PR adds a configuration file during the image-building process so it can be used by the [LimeSurvey docker entrypoint](https://github.com/adamzammit/limesurvey-docker/blob/master/docker-entrypoint.sh#L58) when starting the service to populate its configurations. The base file config.php was copied from a remote installation and then modified by appending:

```
         'request' => array(
			'enableCsrfValidation'=>true,
			'csrfCookie' => array(
				'sameSite' => 'None',
				'secure' => true,
			),
		),
```
and
```
 'x_frame_options' => 'allow',
```
So the LimeSurvey installations allow being embedded in domains different than theirs.

This PR solves this customization issue: https://github.com/eduNEXT/tutor-contrib-limesurvey/issues/6

### How to test
1. Install this `tutor-contrib-limesurvey` version
2. Run `tutor config save`
3. Run `tutor dev start -d`
4. Create a new LimeSurvey component, click `Edit` and add our remote installation under LimeSurvey URL:
![image](https://github.com/eduNEXT/tutor-contrib-limesurvey/assets/64440265/0358ed04-43ba-47d0-8ef7-a8850aba2814)
You should be able to perform any operation when visiting the LimeSurvey tab from the instructor dashboard.
